src/index.ts

```diff
@@ -121,26 +121,39 @@ export const main = async (options: Options): Promise<void> => {
 
 	core.info(`S3 prefix: ${keyPrefix}`);
 
-	await sync(store, stagingDir, 'riffraff-artifact', keyPrefix);
-
-	// Do this bit last to avoid any race conditions, as this is the file that
-	// triggers RR CD.
-	await store.put(
-		Buffer.from(manifestJSON, 'utf8'),
-		'riffraff-builds',
-		keyPrefix + '/build.json',
-	);
+	try {
+		await sync(store, stagingDir, 'riffraff-artifact', keyPrefix);
+
+		// Do this bit last to avoid any race conditions, as this is the file that
+		// triggers RR CD.
+		await store.put(
+			Buffer.from(manifestJSON, 'utf8'),
+			'riffraff-builds',
+			keyPrefix + '/build.json',
+		);
 
-	core.info('Upload complete.');
+		core.info('Upload complete.');
 
-	const pullRequestNumber = await getPullRequestNumber(pullRequestComment);
-	if (pullRequestNumber) {
-		core.info(`Commenting on PR ${pullRequestNumber}`);
-		await commentOnPullRequest(pullRequestNumber, pullRequestComment);
-	} else {
-		core.info(
-			`Unable to calculate Pull Request number, so cannot add a comment. Event is ${context.eventName}`,
+		const pullRequestNumber = await getPullRequestNumber(pullRequestComment);
+		if (pullRequestNumber) {
+			core.info(`Commenting on PR ${pullRequestNumber}`);
+			await commentOnPullRequest(pullRequestNumber, pullRequestComment);
+		} else {
+			core.info(
+				`Unable to calculate Pull Request number, so cannot add a comment. Event is ${context.eventName}`,
+			);
+		}
```

Comment:
Does this need to be in the try block?

Reply:
Ah, technically no, however given we're catching a generic exception (and re-throwing) I think it's OK for now?